### PR TITLE
APID-455 - GET /cmb/box endpoint without boxname in payload

### PR DESCRIPTION
**Pull request by sivaprakashiv**

### CODE REVIEW

File: app/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxController.scala [hunk 1]
```diff
@@ -98,10 +98,16 @@ class BoxController @Inject()(validateUserAgentHeaderAction: ValidateUserAgentHe
     }
   }
 
-  def getBoxesByClientId(): Action[AnyContent] = (Action andThen validateAcceptHeaderAction andThen authAction).async {
+  def getClientManagedBoxesByClientId(): Action[AnyContent] = (Action andThen validateAcceptHeaderAction andThen authAction).async {
```
Comment (ghchapman) [inline app/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxController.scala:101]: The previous version was actually correct - the boxes this method returns may or may not be client managed.

File: app/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxController.scala [hunk 2]
```diff
@@ -98,10 +98,16 @@ class BoxController @Inject()(validateUserAgentHeaderAction: ValidateUserAgentHe
...
     implicit request =>
         boxService.getBoxesByClientId(request.clientId).map { boxes =>
-        Ok(Json.toJson(boxes))
+        Ok(Json.toJson(boxes.map(box =>
+          if(box.clientManaged) {
+            Box(box.boxId, box.boxName, box.boxCreator, box.applicationId, box.subscriber, box.clientManaged)
```
Comment (ghchapman) [inline app/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxController.scala:106]: Can you just return box here, the one you're instantiating is exactly the same.

File: test/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxControllerSpec.scala [hunk 3]
```diff
@@ -446,7 +446,7 @@ class BoxControllerSpec extends AsyncHmrcSpec with GuiceOneAppPerSuite with Befo
         when(mockBoxService.getBoxesByClientId(eqTo(clientId))(*)).thenReturn(Future.successful(List(box)))
         val result = doGet("/cmb/box", validHeaders)
         val expected =
-          s"""[{"boxId":"$boxIdStr","boxName":"boxName","boxCreator":{"clientId":"$clientIdStr"},"clientManaged":false}]"""
+          s"""[{"boxId":"$boxIdStr","boxName":"DEFAULT","boxCreator":{"clientId":"$clientIdStr"},"clientManaged":false}]"""
```
Comment (ghchapman) [inline test/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxControllerSpec.scala:449]: Do we have a unit test exercising the clientManaged==true branch?

Comment (sivaprakashiv) [reply test/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxControllerSpec.scala:449]: Yes, we have one

**Review by ghchapman — APPROVED** [review]

**Pull request merged**